src/pykeen/datasets/ogb.py

```diff
@@ -150,6 +174,155 @@ class OGBWikiKG2(OGBLoader):
 
     name = "ogbl-wikikg2"
 
+    # docstr-coverage: inherited
+    def _load_mappings(self, mapping_root: pathlib.Path) -> tuple[EntityMapping, RelationMapping]:  # noqa: D102
+        df_ent = pandas.read_csv(mapping_root.joinpath("nodeidx2entityid.csv.gz"))
+        entity_to_id = dict(zip(df_ent["entity id"].tolist(), df_ent["node idx"].tolist()))
+        df_rel = pandas.read_csv(mapping_root.joinpath("reltype2relid.csv.gz"))
+        relation_to_id = dict(zip(df_rel["rel id"].tolist(), df_rel["reltype"].tolist()))
+        return entity_to_id, relation_to_id
+
+    # docstr-coverage: inherited
+    def _load_data_dict_for_split(self, dataset, which):
+        # noqa: D102
+        data_dict = torch.load(
+            pathlib.Path(dataset.root).joinpath("split", dataset.meta_info["split"], which).with_suffix(".pt")
+        )
+        if which == "train":
+            data_dict = cast(WikiKG2TrainDict, data_dict)
+        else:
+            data_dict = cast(WikiKG2EvalDict, data_dict)
+        return data_dict
+
+    # docstr-coverage: inherited
+    def _compose_mapped_triples(self, data_dict: WikiKG2TrainDict | WikiKG2EvalDict) -> numpy.ndarray:  # noqa: D102
+        return numpy.stack([data_dict["head"], data_dict["relation"], data_dict["tail"]], axis=-1)
+
+
+#: the node types
+OGBBioKGNodeType = Literal["disease", "drug", "function", "protein", "sideeffect"]
+NODE_TYPES = typing.get_args(OGBBioKGNodeType)
+
+
+class BioKGTrainDict(WikiKG2TrainDict):
+    """A type hint for dictionaries of OGB preprocessed training triples for BioKG."""
+
+    # shape: (n,)
+    head_type: Sequence[OGBBioKGNodeType]
+    tail_type: Sequence[OGBBioKGNodeType]
+
+
+class BioKGEvalDict(BioKGTrainDict):
+    """A type hint for dictionaries of OGB preprocessed evaluation triples for BioKG."""
+
+    # dtype: numpy.int64, shape: (n, k)
+    head_neg: numpy.ndarray
+    tail_neg: numpy.ndarray
+
+
+def load_partial_entity_mapping(mapping_root: pathlib.Path, node_type: OGBBioKGNodeType) -> pandas.DataFrame:
+    """Load a partial entity mapping for a single node type."""
+    # disease: UMLS CUI (https://www.nlm.nih.gov/research/umls/index.html).
+    # drug: STITCH ID (http://stitch.embl.de/).
+    # function: Gene Ontology ID (http://geneontology.org/).
+    # protein: Proteins: Entrez Gene ID (https://www.genenames.org/).
+    # side effects: UMLS CUI (https://www.nlm.nih.gov/research/umls/index.html).
+    # todo(@cthoyt): proper prefixing?
```

Comment:
@cthoyt the labels are from these namespaces; so far I only prepended the node type, but there may be a more useful way how to normalize/prefix those.

Reply:
https://github.com/pykeen/pykeen/pull/1253#pullrequestreview-1386516325